docassemble/AssemblyLine/sessions.py

```diff
@@ -381,13 +485,211 @@ def interview_list_html(
     return table
 
 
+def nice_interview_title(
+    answer: Dict[str, str],
+) -> str:
+    """
+    Return a human readable version of the interview name. Will try several strategies
+    in descending priority order.
+    1. Try looking up the interview title from the `dispatch` directive
+    1. Try removing the package and path from the filename and replace _ with spaces.
+    4. Finally, return "Untitled interview" or translated phrase from system-wide words.yml
+    """
+    if answer.get("filename"):
+        for interview in system_interviews:
+            if answer.get("filename") == interview.get("filename") and interview.get(
+                "title"
+            ):
+                return interview["title"]
+        filename = os.path.splitext(os.path.basename(answer.get("filename", "")))[0]
+        if ":" in filename:
+            filename = filename.split(":")[1]
+        return filename.replace("_", " ").capitalize()
+    else:
+        return word("Untitled interview")
+
+
+def nice_interview_subtitle(answer: Dict[str, str]):
+    """
+    Return either the "title" metadata, or the results of nice_interview_title if undefined
+    """
+    if answer.get("title"):
+        return answer.get("title")
+    else:
+        return nice_interview_title(answer)
+
+
+def radial_progress(answer: Dict[str, Union[str, int]]):
+    """
+    Return HTML for a radial progress bar, or the number of steps if progress isn't available in the metadata.
+    """
+    if not answer.get("progress"):
+        return f"Page {answer.get('steps') or answer.get('num_keys') or 1}"
+
+    # For simulation purposes, assume a form is complete at page 30
+    progress: int = (
+        int(answer["progress"])
+        or int(answer["steps"])
+        or (int(answer["num_keys"]) or 1) * 4
+    )
+
+    # Get a number divisible by 10 and between 0 and 100 (percentage)
+    nearest_10 = min(max(round(progress / 10) * 10, 0), 100)
+    return f"""
+      <div class="radialProgressBar progress-{ nearest_10 }">
+          <div class="overlay">{ nearest_10 }%</div>
```

Comment:
Great point, I'll add the aria.

To get the radial progress bar to display, you need to add the metadata item "progress" to one of your answers. There's no current way that happens automatically. I'm considering that feature "latent" until I have time to test if adding those SQL queries on every page load slows down interviews too much.

Reply:
I guess it might be fine. Forgot that it would be in a column with "progress" already (and that part shouldn't be in the widget anyway). I guess "10%" by itself will be okay.